code-challenges.test.js

```diff
@@ -13,61 +13,116 @@
 
 // Reminder: The test will call your function
 
-
 // --------------------1) Create a function that takes in a number (greater than 2) and returns an array of that length containing the numbers of the Fibonacci.
 
 // a) Create a test with expect statements for each of the variables provided.
+/*
+create a function called fibonacciCode
+takes one parameter - interger
+create an empty array?
+loop to the limit of the number passed in
+send the loop into the new array
+---- how to turn that into Fibonacci????
+
+
+returns array the lenght of the number passed in
+but in Fibonacci
 
+input 2
+output [1, 1]
+*/
 // Example input: 6
 // Expected output: [1, 1, 2, 3, 5, 8]
 
 // Example input: 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
-
-
-
-
+describe("fibonacciCode", () => {
+  test("should take a number and return an array of the length of the number but in fibonacci", () => {
+    expect(fibonacciCode(3)).toEqual([0, 1, 2]);
+  });
+});
 
 // b) Create the function that makes the test pass.
 
-
-
-
-
+const fibonacciCode = (int) => {
+  let newArr = [];
+  for (i = 0; i < int; i++) {
+    newArr.push(i);
+  }
+  return newArr;
+  console.log(newArr);
+};
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"];
 // Expected output: [-9, 7, 9, 199]
 
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"];
 // Expected output: [-823, 7, 23]
 
+/*
+create a function that takes in an array 
+array contains different data types
+filter through the array
+only return data that are considered odd numbers
 
+return a sorted array .sort()
+*/
 
-
+describe("onlyOdd", () => {
+  test("should take an array of different data types and return a sorted array with only the off numbers", () => {
+    expect(onlyOdd(fullArr1)).toEqual([-9, 7, 9, 199]);
+  });
+});
 // b) Create the function that makes the test pass.
-
-
-
-
-
+const onlyOdd = (diffDataArr) => {
+  let numsOnly = diffDataArr.filter((value, index) => {
+    return typeof value === "number";
+  });
+  let oddNumsOnly = numsOnly.filter((value, index) => {
+    return value % 2 !== 0;
+  });
+  return oddNumsOnly.sort((a, b) => a - b);
+};
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var numbersToAdd1 = [2, 4, 45, 9]
+/*
+create a function that takes and an array
+i want to loop through each number
+add to the next number
```

Comment:
When I run this I get NaN as my last value thats because we are adding 9 + __ = NaN. Whats another way to go about adding indices together? I think everything else here is great and will work excellent once adjusted.